src/mon/Paxos.h

```diff
@@ -1339,10 +1352,16 @@ class Paxos {
    *
    * @param c A callback
    */
-  void wait_for_writeable(Context *c) {
+  void wait_for_writeable(MonOpRequestRef op, Context *c) {
     assert(!is_writeable());
+    if (op)
+      op->mark_event("paxos:wait_for_writeable");
     waiting_for_writeable.push_back(c);
   }
+  void wait_for_writeable(Context *c) {
+    MonOpRequestRef o;
+    wait_for_writeable(o, c);
+  }
```

Comment:
I'm confused; how does this make any sense to add a MonOpRequestRef if the context isn't already associated with one?